src/hooks/battery/useBattery.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from 'react';
+
+declare global {
+  interface BatteryManager extends EventTarget {
+    charging: boolean;
+    chargingTime: number;
+    dischargingTime: number;
+    level: number;
+  }
+
+  interface Navigator {
+    getBattery?: () => Promise<BatteryManager>;
+  }
+}
+
+const noop = () => {};
+
+/**
+ * Interfaces with the Battery Management API (if available; at present only in Chromium-based browsers)
+ * to return the current battery charge level and charging status.
+ */
+const useBattery = () => {
+  const [chargePercent, setChargePercent] = useState<number>(100);
+  const [isCharging, setIsCharging] = useState<boolean>(true);
```

Comment:
By default I think we'll want to set the default state to be `'getBattery' in navigator`, otherwise browsers like iOS and FireFox that don't support the battery API will always show the plug icon

Reply:
Unfortunately I don't believe we'll be able to set the initial state in this way since the battery API is completely absent on those browsers. Maybe we can instead change this to `false` so that the indicator defaults to its current appearance, i.e. fully-charged but not charging?

Reply:
`false` is probably a better solution here 👍  

But just so you know, `'getBattery' in navigator` is simply checking that a key exists within an object and doesn't require the Battery API to exist. In this case we'd check the `navigator` object for the `getBattery` function since that function will only exist if the browser supports the Battery API